avocado/plugins/jobs.py

```diff
@@ -0,0 +1,238 @@
+                   "experimental and only supported when job executed with "
+                   "the Spawner architecture.")
+            print(msg.format(job_id))
+            return exit_codes.AVOCADO_GENERIC_CRASH
+
+        for test in results_data.get('tests'):
+            test_id = test.get('id')
+            print("Downloading files for test {} ...".format(test_id))
+            try:
+                files_buffers = spawner().stream_output(job_id, test_id)
+                for filename, stream in files_buffers:
+                    destination = os.path.join(destination, filename)
+                    self._save_stream_to_file(stream, destination)
+            except SpawnerException as ex:
+                print("Error: Failed to download: {}. Exiting...".format(ex))
+                return exit_codes.AVOCADO_GENERIC_CRASH
+        return exit_codes.AVOCADO_ALL_OK
+
+    def handle_show_command(self, config):
+        """Called when 'avocado jobs show' command is executed."""
+
+        job_id = config.get('jobs.show.job_id')
+        results_dir = get_job_results_dir(job_id)
+        results_file = os.path.join(results_dir, 'results.json')
+        config_file = os.path.join(results_dir, 'jobdata/args.json')
+        try:
+            results_data = self._get_data_from_file(results_file)
+        except FileNotFoundError as ex:
+            # Results data are important and should exit if not found
+            print(ex)
+            return exit_codes.AVOCADO_GENERIC_CRASH
+
+        try:
+            config_data = self._get_data_from_file(config_file)
+        except FileNotFoundError:
+            pass
+
+        data = {'Job id': job_id,
+                'Debug log': results_data.get('debuglog'),
+                'Spawner': config_data.get('nrun.spawner', 'unknown'),
+                '#total tests': results_data.get('total'),
+                '#pass tests': results_data.get('pass'),
+                '#skip tests': results_data.get('skip'),
+                '#errors tests': results_data.get('errors'),
+                '#cancel tests': results_data.get('cancel')}
+
+        # We could improve this soon with more data and colors
+        self._print_job_details(data)
+        self._print_job_tests(results_data.get('tests'))
+        return exit_codes.AVOCADO_ALL_OK
+
+    def run(self, config):
+        results = {}
+
+        jobs_dir = get_logs_dir()
+        for result in glob(os.path.join(jobs_dir, '*/results.json')):
+            with open(result, 'r') as fp:
+                job = json.load(fp)
+                results[job['job_id']] = result
+
+        subcommand = config.get('jobs_subcommand')
+        if subcommand == 'list':
+            return self.handle_list_command(results)
+        elif subcommand == 'show':
+            return self.handle_show_command(config)
+        elif subcommand == 'get-output-files':
+            return self.handle_output_files_command(config)
+        return exit_codes.AVOCADO_ALL_OK
```

Comment:
I know there are lots of opinions about single versus multiple return statements in a method, but I think we have a good opportunity for readability here using a single return. Check if code here helps:

https://github.com/avocado-framework/avocado/blob/master/avocado/plugins/assets.py#L321-L350

Reply:
I think that is questionable. ;) When using the return there it is clear that we don't have to read the lines below, because we know that the code "stops there". So IMO, this is easy to read then introducing a new variable and making the code continue the execution.

Reply:
> I think that is questionable. ;) When using the return there it is clear that we don't have to read the lines below, because we know that the code "stops there". So IMO, this is easy to read then introducing a new variable and making the code continue the execution.

I agree with you if we consider this specific small block of code. For long blocks of code, where sometimes you may just want to fix a bug in the middle of it, having multiple points of return may be more confusing than helping to debug.

So, here it is up to you as the code is small and easy to read in one screen.